airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/data/AirbyteValueIdentityMapper.kt

```diff
@@ -8,20 +8,47 @@ import io.airbyte.cdk.load.message.DestinationRecord
 import io.airbyte.protocol.models.v0.AirbyteRecordMessageMetaChange.Change
 import io.airbyte.protocol.models.v0.AirbyteRecordMessageMetaChange.Reason
 
-open class AirbyteValueIdentityMapper(
-    val meta: DestinationRecord.Meta,
-) {
+interface AirbyteValueMapper {
+    val collectedChanges: List<DestinationRecord.Change>
+
+    fun map(
+        value: AirbyteValue,
+        schema: AirbyteType,
+        path: List<String> = emptyList(),
+    ): AirbyteValue
+}
+
+/** An optimized identity mapper that just passes through. */
+class AirbyteValueNoopMapper : AirbyteValueMapper {
+    override val collectedChanges: List<DestinationRecord.Change> = emptyList()
+
+    override fun map(
+        value: AirbyteValue,
+        schema: AirbyteType,
+        path: List<String>,
+    ): AirbyteValue = value
+}
+
+open class AirbyteValueIdentityMapper : AirbyteValueMapper {
+    override val collectedChanges: List<DestinationRecord.Change>
+        get() = changes.toList().also { changes.clear() }
```

Comment:
this feels hard to reason about (e.g. this is clearly stateful + not thread safe, so the caller has to remember that)

does anything break if we just shove all the state into the function directly?
```kotlin
fun map(...): Pair<AirbyteValue, List<Change>> {
  val collectedChanges = mutableListOf()
  fun collectFailure(...) { ... }
  fun mapImpl(...): AirbyteValue = // your existing map code
  val result = mapImpl(...)
  return Pair.of(result, collectedChanges)
}
```